src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -161,7 +167,7 @@ public bool IsLegacyCSProjPackageReferenceProject
                             _isLegacyCSProjPackageReferenceProject = true;
                         }
                     }
-                    else if (_context.NuGetProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase))
+                    else if (restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase))
```

Comment:
If this is true but it isn't a VSProject4 I don't think this should be true.  I would instead put this check around the final true, if everything else matches then either of the following must be true.

1. RestoreProjectStyle is not set
1. RestoreProjectStyle is PackageReference

If you have overridden this to be packages.config for example you don't want this project system.